hikey_debian.mk

```diff
@@ -123,26 +124,38 @@ arm-tf-clean:
 ################################################################################
 # EDK2 / Tianocore
 ################################################################################
-EDK2_VARS ?= EDK2_ARCH=AARCH64 \
-		EDK2_DSC=HisiPkg/HiKeyPkg/HiKey.dsc \
-		EDK2_TOOLCHAIN=GCC49 \
-		EDK2_BUILD=$(EDK2_BUILD)
+EDK2_ARCH ?= AARCH64
+EDK2_DSC ?= OpenPlatformPkg/Platforms/Hisilicon/HiKey/HiKey.dsc
+EDK2_TOOLCHAIN ?= GCC49
 
 EDK2_CONSOLE_UART ?= $(CFG_NW_CONSOLE_UART)
 ifeq ($(EDK2_CONSOLE_UART),0)
-	EDK2_VARS += EDK2_MACROS="-DSERIAL_BASE=0xF8015000"
+	EDK2_BUILDFLAGS += -DSERIAL_BASE=0xF8015000
 endif
 
 define edk2-call
 	GCC49_AARCH64_PREFIX=$(LEGACY_AARCH64_CROSS_COMPILE) \
-	$(MAKE) -j1 -C $(EDK2_PATH) \
-		-f HisiPkg/HiKeyPkg/Makefile $(EDK2_VARS)
+	build -n 1 -a $(EDK2_ARCH) -t $(EDK2_TOOLCHAIN) -p $(EDK2_DSC) \
+		-b $(EDK2_BUILD) $(EDK2_BUILDFLAGS)
 endef
 
-edk2: edk2-common
+.PHONY: edk2
+edk2:
+	cd $(EDK2_PATH) && rm -rf OpenPlatformPkg && \
```

Comment:
https://github.com/96boards-hikey/edk2 hikey-aosp branch has a git submodule to https://github.com/96boards-hikey/OpenPlatformPkg hikey-aosp branch, i.e. at an old commit, so when we `repo sync edk2`, `OpenPlatformPkg` exists as a dir in `edk2/`, not a symlink, so `ln -sf` or link by manifest won't work. I just followed the rpb build procedure [1], i.e. rm the dir and symlink to the one repo sync'ed separately in the build root dir.

I suppose we can try submitting a PR to https://github.com/96boards-hikey/edk2 to remove the submodule? That way we don't have to `rm -rf OpenPlatformPkg` and can just `ln -sf` directly.

[1] https://git.linaro.org/ci/job/configs.git/tree/rpb-uefi/stable/builders.sh#n76